%\VignetteEngine{knitr}
%\VignetteIndexEntry{RforProteomics Bioc2013 poster}
%\VignetteKeywords{bioinformatics, proteomics, mass spectrometry}

\documentclass[final]{beamer} 

\mode<presentation> {  %% check http://www-i6.informatik.rwth-aachen.de/~dreuw/latexbeamerposter.php for examples
  %% \usetheme{CCP}
}

\setbeamertemplate{bibliography item}[text]

\boldmath
\usepackage[orientation=portrait,size=a0,scale=1.4,debug]{beamerposter}                        % e.g. for DIN-A0 poster
%\usepackage[orientation=portrait,size=a1,scale=1.4,grid,debug]{beamerposter}                  % e.g. for DIN-A1 poster, with optional grid and debug output
%\usepackage[size=custom,width=200,height=120,scale=2,debug]{beamerposter}                     % e.g. for custom size poster
%\usepackage[orientation=portrait,size=a0,scale=1.0,printer=rwth-glossy-uv.df]{beamerposter}   % e.g. for DIN-A0 poster with rwth-glossy-uv printer check
% ...
%

%% hide navigation symbols (bottom right)
\setbeamertemplate{navigation symbols}{}

\usepackage{ragged2e} 
\usepackage[english]{babel}
\usepackage[latin1]{inputenc}
\usepackage{amsmath,amsthm, amssymb, latexsym}
\usefonttheme[onlymath]{serif}

\usepackage{graphicx}
\usepackage{caption}
\usepackage{subcaption}

\usepackage{tcolorbox}

\newcommand{\E}{\mathsf{E}}
\newcommand{\VAR}{\mathsf{VAR}}
\newcommand{\COV}{\mathsf{COV}}
\newcommand{\Prob}{\mathsf{P}}

\newcommand{\R}{\texttt{R} }
\newcommand{\code}[1]{{\texttt{#1}}}
\newcommand{\Rfunction}[1]{{\texttt{#1}}}
\newcommand{\Robject}[1]{{\texttt{#1}}}
\newcommand{\Rpackage}[1]{{\mbox{\texttt{#1}}}}
\newcommand{\email}[1]{\href{mailto:#1}{\normalfont\texttt{#1}}}

\newcommand{\challenge}[1]{
       \begin{tcolorbox}[notitle,boxrule=0pt,colback=blue!10,colframe=blue!10]\textbf{Challenges}~{#1}
       \end{tcolorbox}
}


%% colors
\definecolor{Red}{rgb}{0.7,0,0}
\definecolor{Blue}{rgb}{0,0,0.8}


\usepackage[bordercolor=white, backgroundcolor=gray!20]{todonotes}

\usepackage{hyperref}
\usepackage{breakurl}
\hypersetup{%
  pdfauthor={Laurent Gatto},%
  pdfusetitle,
  bookmarks = {true},
  bookmarksnumbered = {true},
  bookmarksopen = {true},
  bookmarksopenlevel = 2,
  unicode = {true},
  breaklinks = {false},
  hyperindex = {true},
  colorlinks = {true},
  linktocpage = {true},
  plainpages = {false},
  linkcolor = {Blue},
  citecolor = {Blue},
  urlcolor = {Red},
  pdfstartview = {Fit},
  pdfpagemode = {UseOutlines},
  pdfview = {XYZ null null null}
}



%% figure numering
\usecaptiontemplate{ 
  \small 
  \structure{\insertcaptionname~\insertcaptionnumber:} 
  \insertcaption 
} 

\title[RforProteomics]{\huge Using \R and Bioconductor for proteomics data analysis}

\author[Gatto et al.]{\large L. Gatto$^{*, 1}$, S. Gibb$^2$, L.M. Breckels$^1$, A. Christoforou$^1$ and K. S. Lilley$^1$}

\institute[CCP]{
  \begin{small}
    $^1$Cambridge Centre for Proteomics, Department of Biochemistry, University of Cambridge, UK \\
    $^2$Institute for Medical Informatics, Statistics and Epidemiology, University of Leipzig, Germany \\
    \bigskip
    $^{*}$\url{lg390@cam.ac.uk} -- \url{http://www.bio.cam.ac.uk/proteomics/}
  \end{small}
}

\date[Bioc2013]{Bioconductor Meeting, 18 -- 19 July 2013}
\date[]{}

\begin{document}

<<env, include=FALSE, echo=FALSE, cache=FALSE>>=
library("knitr")
opts_chunk$set(fig.align = 'center', 
               fig.show = 'hold', 
               par = TRUE,
               prompt = FALSE,
               eval = TRUE,
               stop_on_error = 1L,
               comment = ':')
options(replace.assign = TRUE, 
        width = 55)

suppressPackageStartupMessages(library("RforProteomics"))
suppressPackageStartupMessages(library("foreach"))
suppressPackageStartupMessages(library("doMC"))
suppressPackageStartupMessages(library("iterators"))
suppressPackageStartupMessages(library("MSnbase"))
suppressPackageStartupMessages(library("mzR"))
suppressPackageStartupMessages(library("msdata"))
set.seed(1)
@ 
%%$

\begin{frame}[fragile]

  \maketitle

  \begin{columns}
    \begin{column}{.48\textwidth}
      %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
      %% first column %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
      %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
     \begin{block}{Introduction}
       \justifying 

       \R and Bioconductor have had a tremendous impact on the quality of genomics data analysis \cite{Gentleman2004},
       demonstrating that extraction of relevant and biologically meaningful information from high-throughput data, 
       requires investing time and effort in the exploration and analysis of the data.
       
       \bigskip

       Even well-known and respected leaders in proteomics agree that it lies 10 years behind genomics. 
       There are several valid reasons for this, including the chemical complexity of proteins, 
       the technical complexity of the instrumentation (in particular mass-spectrometry - MS) 
       and the vast possibilities in the study of proteins. 
       An often overseen albeit essential component of this failure is arguably the software that is promoted inside the proteomics community. 
       Computational proteomics researchers who value quality software, comprehensive data analysis and 
       reproducible research ought to illustrate how more flexible and advanced tools can effectively be used 
       and demonstrate their advantages.
       Here, we illustrate some examples of proteomics data analysis in \texttt{R}, in particular low level 
       \textbf{raw MS data} manipulation, labelled and label-free \textbf{quantitation} and peptide \textbf{identification},
       taken from the \Rpackage{RforProteomics} package \cite{r4p}.
     \end{block}

     \begin{block}{Working with raw data}
       \justifying 
       The proteomics community has developed a range of data standards and formats for MS data 
       (the latest being \texttt{mzML}) to overcome the shortcomings 
       of closed, binary vendor-specific formats. One of the main projects that implement parsers for the XML-based 
       open formats is the \texttt{C++} proteowizard project \cite{Chambers2012}, which is interfaced by the 
       \Rpackage{mzR} Bioconductor package using the \Rpackage{Rcpp} package.

<<mzR>>=
library("mzR")
fname <- dir(system.file(package="MSnbase",dir="extdata"), full.name=TRUE, pattern="mzXML$")
ms <- openMSfile(fname)
@

The resulting \Robject{ms} object is a file handle that allows fast direct and random access to the 
individual spectra. \Rpackage{mzR} is used by a variety of other packages like 
\Rpackage{xcms}, \Rpackage{MSnbase}, \Rpackage{RMassBank} and \Rpackage{TargetSearch}.

       \bigskip
       
         \challenge{Further improve support of raw MS data and develop the range of supported formats, 
         in particular identification (\texttt{mzIdentML}) and quantitation (\texttt{mzQuantML}) formats.}

     \end{block}
     \bigskip
     
     \begin{block}{Labelled quantitation}
       \justifying 
       The same raw data file can be imported in a convenient higher level container and 
       directly processed, plotted, quantified and normalised with the \Rpackage{MSnbase} \cite{msnbase} software.
<<msnbase, eval = FALSE>>= 
exp <- readMSData(fname, verbose = FALSE)
plot(exp[["X3.1"]], full = TRUE, reporters = iTRAQ4)
set <- quantify(exp, method = "trap", reporters = iTRAQ4, verbose = FALSE, parallel = TRUE)
head(exprs(set), n = 3)
@         
\begin{figure}[h]
  \centering
  \begin{subfigure}[b]{0.55\linewidth}
    \centering
    \includegraphics[width=\linewidth]{./figure/plotspectrum.pdf}
  \end{subfigure}
  \begin{subfigure}[b]{0.44\linewidth}
    \includegraphics[width=\linewidth]{./figure/spikes.pdf} \\
    \includegraphics[width=\linewidth]{./figure/plotmzdelta.pdf}
  \end{subfigure}
  \label{fig:msnbase} 
  \caption{MS$^2$ spectrum of an iTRAQ 4-plex experiment showing the 4 isobaric reporter ions, 
    as produced by \Rfunction{plot} above (left). Peptides of interesting from a spiked-in experiment 
    (top right) and distribution of the \textit{m/z} differences of all MS$^2$ spectra from the same 
    experiment, use as a peptide-spectrum matching quality assessment (bottom right).}
\end{figure}

     \challenge{Although labelled MS$^2$ quantitation is well supported with \Rpackage{MSnbase} 
     and \Rpackage{isobar}, metabolic labelling techniques like $^{15}$N or SILAC still need to be developed.}
     
     \end{block}
     \vfill
    \end{column}

    %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
    %% second column %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
    %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

    \begin{column}{.48\textwidth}
      %% next block %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
      \begin{block}{Label-free quantitation}
        \justifying 
        Label-free quantitation is available in the \Rpackage{xcms} \cite{xcms} and \Rpackage{MALDIquant} \cite{maldiquant} packages. 
        The latter provides a complete pipeline, including 
        baseline subtraction, smoothing, peak detection and alignment using warping functions, 
        handling of replicated measurements as well as allowing spectra with different resolutions.
        \begin{figure}[h]
          %% please find the script to create this figure at
          %% https://github.com/sgibb/MALDIquantExamples/blob/master/createFigure1_color.R
          \includegraphics[width=\textwidth]{./figure/mqplot2.pdf}
          \label{fig:mq}
          \caption{Illustration of the \Rpackage{MALDIquant} pipeline:
            raw spectrum with estimate baseline (A);
            variance-stabilized, smoothed, baseline-corrected spectrum with detected peaks (B);
            fitted warping function for peak alignment (C); 
            four unaligned peaks (D);
            four aligned peaks (E); 
            merged spectrum with discovered and labeled peaks (F).}
        \end{figure}
        
        A complete pipeline for MS$^e$ data independent acquisition, including support for ion mobility separation 
        is available in the \Rpackage{synapter} package \cite{synapter} that, among other, transfers identification between acquisitions 
        to substantially reduce missing values. 
        
        \bigskip        
        \challenge{Application and benchmarking of label-free pipeline on popular Thermo Orbitrap instruments.}
      \end{block}
      %%\vfill
      \begin{block}{Peptide identification}
        \justifying 
        The recently released \Rpackage{rTANDEM} package encapsulates the \texttt{X!Tandem} \cite{tandem}
        search engine in \R and uses the same XML-based parameter files as the native application 
        or dedicated \R parameter object. 
        Result files can be directly parsed and mined in \R.          
<<rtandem, eval = FALSE, tidy = FALSE>>=
xmlres <- rtandem(spectra.mgf, taxon = "yeast", 
                  taxonomy = "taxonomy.xml", 
                  default.parameters = "default-params.xml")
## or xmlres <- tandem(param)
res <- GetResultsFromXML(xmlres)
proteins <- GetProteins(res) ## data.table objects
peptides <- GetPeptides(res) 
@       
        A complete pipeline with support for identification is welcome at the cost of additional 
        development and maintenance time for developers. With support for \texttt{mzIdentML} files, 
        it will become possible to import identification data from most search engines, this 
        facilitating the integration of \R based pipelines with existing tools.        
        
        \bigskip        
        \challenge{Better integration of identification and raw/quantitation data infrastructures.}

      \end{block}
      %%\vfill

      %% next block %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
      \begin{block}{Conclusions and perspectives}
       \justifying 

       The flexibility of the \R environment and the breath of available packages is sometimes 
       daunting for newcomers and introductory points of entry are welcome.  
       The \Rpackage{RforProteomics} package [\url{https://github.com/lgatto/RforProteomics}]
       %% (and the proteomics workflow on the Bioconductor homepage) 
       ought to assume such a role. For this, \Rpackage{RforProteomics} should be a collaborative project 
       and contributions through the github repository are encouraged. 
       
       \bigskip

        Despite well known advantages in terms of statistical analyses of data and some 
        unique software for proteomics and mass-spectrometry data analysis, 
        there remains a lot of efforts and work to be done for \texttt{R}/Bioconductor to become a 
        complete framework for proteomics data processing. These efforts should be 
        tackled by a group of developers. It is our hope that the \Rpackage{RforProteomics}
        will be a helpful targeted introduction to new users and motivate collaborative 
        development of package developers.
      \end{block}
      %%\vfill
      \begin{block}{}
        \justifying
        \tiny
        
        \begin{thebibliography}{9}          

        \bibitem{synapter} Bond NJ \textit{et al.}
          \emph{Improving Qualitative and Quantitative Performance for MS$^E$-based Label-free Proteomics}. 
          J Proteome Res. 2013 %% Jun 7;12(6):2340-53. doi: 10.1021/pr300776t. Epub 2013 Apr 17. 
          PMID: 23510225.

        \bibitem{Chambers2012} Chambers M. \textit{et al.}
          \emph{A cross-platform toolkit for mass spectrometry and proteomics}. 
          Nat Biotechnol. 2012 %% Oct;30(10):918-20. doi: 10.1038/nbt.2377. 
          PMID: 23051804.
          
        \bibitem{tandem} Craig R and Beavis RC. 
          \emph{TANDEM: matching proteins with tandem mass spectra}.
          Bioinformatics. 2004 %% Jun 12;20(9):1466-7. Epub 2004 Feb 19. 
          PMID: 14976030.                  
                              
        \bibitem{r4p} Gatto L and Christoforou A. 
          \emph{Using R and Bioconductor for proteomics data analysis}. 
          Biochim Biophys Acta. 2013 %% May 18. doi:pii: S1570-9639(13)00186-6.10.1016/j.bbapap.2013.04.032. 
          PMID: 23692960.
          
        \bibitem{msnbase} Gatto L and Lilley KS
          \emph{MSnbase-an R/Bioconductor package for isobaric tagged mass spectrometry data visualization, processing and quantitation}.
          Bioinformatics. 2012 %% Jan; 28:(2)288-9
          PMID: 22113085.

        \bibitem{Gentleman2004} Gentleman R. \textit{et al.}
          \emph{Bioconductor: open software development for computational biology and bioinformatics}. 
          Genome Biol. 2004 %% ;5(10):R80. Epub 2004 Sep 15. 
          PMID: 15461798.          
          
        \bibitem{maldiquant} Gibb S and Strimmer K
          \emph{MALDIquant: a versatile R package for the analysis of mass spectrometry data.}
          Bioinformatics. 2012 %% Sep; 28:(17)2270-1           
          PMID: 22796955.
          
        \bibitem{xcms} Smith CA \textit{et al.}
          \emph{XCMS: Processing mass spectrometry data for metabolite profiling using nonlinear peak alignment, matching and identification}.
          Analytical Chemistry 2006 %% 78:779-787          
          PMID: 16448051.                                        
            
        \end{thebibliography}        
      \end{block}
      
      %% Funding %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
      %% \vspace{3cm}
      \begin{columns}[T]
        \begin{column}{.15\textwidth}
          \includegraphics[width=\linewidth]{./figure/PRIME-XS_Logo}
        \end{column}
        \begin{column}{.85\textwidth}
          \small This work has been supported by the PRIME-XS project, grant agreement number 262067, 
          funded by the European Union 7$^{th}$ Framework Program. 
          S.G. received funding from the German National Academic Foundation.
        \end{column}
      \end{columns}
      %% Funding %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
      
      %% \vspace{2mm}
    \end{column}
  \end{columns}

\end{frame}


\end{document}

