%\VignetteEngine{knitr}
%\VignetteIndexEntry{RforProteomics poster}
%\VignetteKeywords{bioinformatics, proteomics, mass spectrometry}

\documentclass[final]{beamer} 

\mode<presentation> {  %% check http://www-i6.informatik.rwth-aachen.de/~dreuw/latexbeamerposter.php for examples
  %% \usetheme{CCP}
}

\boldmath
\usepackage[orientation=portrait,size=a0,scale=1.4,debug]{beamerposter}                        % e.g. for DIN-A0 poster
%\usepackage[orientation=portrait,size=a1,scale=1.4,grid,debug]{beamerposter}                  % e.g. for DIN-A1 poster, with optional grid and debug output
%\usepackage[size=custom,width=200,height=120,scale=2,debug]{beamerposter}                     % e.g. for custom size poster
%\usepackage[orientation=portrait,size=a0,scale=1.0,printer=rwth-glossy-uv.df]{beamerposter}   % e.g. for DIN-A0 poster with rwth-glossy-uv printer check
% ...
%
\usepackage{ragged2e} 
\usepackage[english]{babel}
\usepackage[latin1]{inputenc}
\usepackage{amsmath,amsthm, amssymb, latexsym}
\usefonttheme[onlymath]{serif}

\newcommand{\E}{\mathsf{E}}
\newcommand{\VAR}{\mathsf{VAR}}
\newcommand{\COV}{\mathsf{COV}}
\newcommand{\Prob}{\mathsf{P}}

\newcommand{\R}{\texttt{R} }
\newcommand{\code}[1]{{\texttt{#1}}}
\newcommand{\Rfunction}[1]{{\texttt{#1}}}
\newcommand{\Robject}[1]{{\texttt{#1}}}
\newcommand{\Rpackage}[1]{{\mbox{\texttt{#1}}}}
\newcommand{\email}[1]{\href{mailto:#1}{\normalfont\texttt{#1}}}

%% colors
\definecolor{Red}{rgb}{0.7,0,0}
\definecolor{Blue}{rgb}{0,0,0.8}

\usepackage{hyperref}
\usepackage{breakurl}
\hypersetup{%
  pdfauthor={Laurent Gatto},%
  pdfusetitle,
  bookmarks = {true},
  bookmarksnumbered = {true},
  bookmarksopen = {true},
  bookmarksopenlevel = 2,
  unicode = {true},
  breaklinks = {false},
  hyperindex = {true},
  colorlinks = {true},
  linktocpage = {true},
  plainpages = {false},
  linkcolor = {Blue},
  citecolor = {Blue},
  urlcolor = {Red},
  pdfstartview = {Fit},
  pdfpagemode = {UseOutlines},
  pdfview = {XYZ null null null}
}



%% figure numering
\usecaptiontemplate{ 
  \small 
  \structure{\insertcaptionname~\insertcaptionnumber:} 
  \insertcaption 
} 

\title[RforProteomics]{\huge Using \R and Bioconductor for proteomics data analysis}

\author[Gatto et al.]{\large L. Gatto$^{*, 1}$, L.M. Breckels$^1$, S. Gibb$^2$, A. Christoforou$^1$ and K. S. Lilley$^1$}

\institute[CCP]{
  \begin{small}
    $^1$Cambridge Centre for Proteomics, Department of Biochemistry, University of Cambridge, UK \\
    $^2$Institut f\"ur Medizinische Informatik, Statistik und Epidemiologie, Universit\"at Leipzig, Germany \\
    \bigskip
    $^{*}$\url{lg390@cam.ac.uk} -- \url{http://www.bio.cam.ac.uk/proteomics/}
  \end{small}
}

\date[Bioc2013]{Bioconductor Meeting, 18 -- 19 July 2013}
\date[]{}

\begin{document}

<<env, include=FALSE, echo=FALSE, cache=FALSE>>=
library("knitr")
opts_chunk$set(fig.align = 'center', 
               fig.show = 'hold', 
               par = TRUE,
               prompt = FALSE,
               eval = TRUE,
               stop_on_error = 1L,
               comment = ':')
options(replace.assign = TRUE, 
        width = 55)

suppressPackageStartupMessages(library("RforProteomics"))
suppressPackageStartupMessages(library("foreach"))
suppressPackageStartupMessages(library("doMC"))
suppressPackageStartupMessages(library("iterators"))
suppressPackageStartupMessages(library("MSnbase"))
suppressPackageStartupMessages(library("mzR"))
suppressPackageStartupMessages(library("msdata"))
set.seed(1)
@ 
%%$

\begin{frame}[fragile]
  \maketitle
  \vfill   

  \begin{columns}
    \begin{column}{.48\textwidth}
      %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
      %% first column %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
      %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
     \begin{block}{Introduction}
       \justifying 

       \R and Bioconductor have had a tremendous impact on the quality of genomics data analysis \cite{Gentleman2004},
       demonstrating that extraction of relevant and biologically meaningful information from high-throughput data, 
       required investing time and effort in the exploration and analysis of the data.
       
       \bigskip

       Even well-known and respected leader in proteomics agree that it lies 10 years behind genomics. 
       There are several valid reasons for this, including the chemical complexity of proteins, 
       the technical complexity of the instrumentation (in particular mass-spectrometry - MS) 
       and the vast possibilities in the study of proteins. 
       An often overseen albeit essential component of this failure is arguably the software that is promoted inside the proteomics community. 
       Computational proteomics researcher who value quality software, comprehensive data analysis and 
       reproducible research ought to illustrate how more flexible and advanced tools can effectively be used 
       and demonstrate their advantages.
       Here, we illustrate some examples of proteomics data analysis in \R.
     \end{block}

     \begin{block}{Working with raw data}
       \justifying 
       The proteomics community has developed a range of data standards and formats for MS data 
       (the latest being \texttt{mzML}) to overcome the shortcomings 
       or closed, binary vendor-specific formats. One of the main projects that implement parsers for the XML-based 
       open formats is the \texttt{C++} proteowizard project \cite{Chambers2012}, which is interfaced by the 
       \Rpackage{mzR} Bioconductor package using the \Rpackage{Rcpp} package.

<<mzR>>=
library("mzR")
fname <- dir(system.file(package="MSnbase",dir="extdata"), full.name=TRUE, pattern="mzXML$")
ms <- openMSfile(fname)
@

The resulting \Robject{ms} object is a file handle that allows fast direct and random access to the 
individual spectra. \Rpackage{mzR} is used by a variety of other packages like 
\Rpackage{xcms}, \Rpackage{MSnbase}, \Rpackage{RMassBank} and \Rpackage{TargetSearch}.

       \bigskip

       \textbf{Challenges} Further improve support of raw MS data and develop the range of supported formats, 
       in particluar identification (\texttt{mzIdentML}) and quantitation (\texttt{mzQuantML}) formats. 
     \end{block}
     \bigskip


     \begin{block}{Labelled quantitation}
       \justifying 
       The same raw data file can be imported in a convenient higher level container and 
       directly processed, plotted, quantified and normalised with the \Rpackage{MSnbase} software. 
<<msnbase, cache=TRUE, warnings = FALSE>>=
exp <- readMSData(fname, verbose = FALSE)
set <- quantify(exp, method = "trap", reporters = iTRAQ4, verbose = FALSE, parallel = TRUE)
head(exprs(set), n = 3)
@        
      \begin{figure}[h]
<<plotspectrum, dev = 'pdf', fig.width = 12, fig.height = 5>>=
plot(exp[["X3.1"]], full = TRUE, reporters = iTRAQ4)
@         
\caption{MS$^2$ spectrum of an iTRAQ 4-plex experiment showing the 4 isobaric reporter ions.}
\label{fig:spectrum}
      \end{figure}
\textbf{Challenges} Altough labelled MS$^2$ quantitation is well supported with \Rpackage{MSnbase} 
and \Rpackage{isobar}, metabolic labelling techniques like N$^{15}$ or SILAC still need to be developed. 
     \end{block}

     \bigskip 

     %% next block %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
     \vspace{3cm}
     \begin{columns}
       \begin{column}{.15\textwidth}
         \includegraphics[width=\linewidth]{./figure/PRIME-XS_Logo}
       \end{column}
       \begin{column}{.85\textwidth}
         \small This work has been supported by the PRIME-XS project, grant agreement number 262067, 
         funded by the European Union 7$^{th}$ Framework Program.
       \end{column}
     \end{columns}
    \end{column}

    %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
    %% second column %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
    %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

    \begin{column}{.48\textwidth}
      %% next block %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
      \begin{block}{Label-free quantitation}
        \justifying 
      \end{block}
      \vfill
      \begin{block}{MS$^e$ data independent acquisition}
        \justifying 
      \end{block}
      \vfill
      \begin{block}{Peptide identification}
        \justifying 
      \end{block}
      \vfill
      %% figures  %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

      \vfill
      %% next block %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
      \begin{block}{Conclusions and perspectives}
       \justifying 

       The felxibility of the \R environment and the breath of available packages is sometimes 
       daunting for newcomers and introductionary points of entry are welcome.  
       The \Rpackage{RforProteomics} package \cite{r4p} [\url{https://github.com/lgatto/RforProteomics}]
       %% (and the proteomics workflow on the Bioconductor homepage) 
       ought to assume such roles. For this, \Rpackage{RforProteomics} should be a collaborative project 
       and contribution through the github repository are encouraged. 
       
       \bigskip

        Despite well known advantages in terms of statistical analyses of data and some 
        unique software for proteomics and mass-spectrometry data analysis, 
        there remains a lot of efforts and work to be done for \texttt{R}/Bioconductor to become a 
        complete framework for proteomics data processing. These efforts should be 
        tackled by a group of developers. It is our hope that the \Rpackage{RforProteomics}
        will be a helpful targeted introduction to new users and motivate collaborative 
        development of package developers.
      \end{block}
      \vfill
      \begin{block}{}
        \justifying
        \tiny
        
        \begin{thebibliography}{9}

        \bibitem{Gentleman2004}
          Gentleman \textit{et al.}
          \emph{Bioconductor: open software development for computational biology and bioinformatics}. 
          Genome Biol. 2004 %% ;5(10):R80. Epub 2004 Sep 15. 
          PMID: 15461798.          
          
        \bibitem{Chambers2012}
          Chambers \textit{et al.}
          \emph{A cross-platform toolkit for mass spectrometry and proteomics}. 
          Nat Biotechnol. 2012 %% Oct;30(10):918-20. doi: 10.1038/nbt.2377. 
          PMID: 23051804.
          
        \bibitem{r4p}
          Gatto L, Christoforou A. 
          \emph{Using R and Bioconductor for proteomics data analysis}. 
          Biochim Biophys Acta. 2013 %% May 18. doi:pii: S1570-9639(13)00186-6.10.1016/j.bbapap.2013.04.032. 
          PMID: 23692960.
          
        \end{thebibliography}
        
      \end{block}

      %% \vspace{2mm}
    \end{column}
  \end{columns}

\end{frame}


\end{document}

